torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -20,7 +20,7 @@ EmbeddingImpl::EmbeddingImpl(EmbeddingOptions options)
 void EmbeddingImpl::reset() {
   weight = register_parameter(
       "weight", torch::empty({options.count_, options.dimension_}));
-  weight.data().normal_(0, 1);
+  weight.normal_(0, 1);
```

Comment:
I put a `NoGradGuard` around it now

Reply:
I think @zdevito said this doesn't prevent the not-a-leaf checks? Or I'm confused about when this happens and when it doesn't.

Reply:
Oh, I see, so if you have a NoGradGuard, the graph is never recorded, so you don't have this issue. This means `NoGradGuard g; weight.zero_()` and `weight.detach().zero_()` are equivalent, except that the second incantation creates a graph node that's immediately destroyed since the result is never assigned to anything.